src/Raven.Client/Documents/Indexes/AdditionalAssembly.cs

```diff
@@ -0,0 +1,188 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AbstractIndexCreationTask.cs" company="Hibernating Rhinos LTD">
+//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Indexes
+{
+    public class AdditionalAssembly : IDynamicJson
+    {
+        public string AssemblyName { get; private set; }
+
+        public string AssemblyPath { get; private set; }
+
+        public string PackageName { get; private set; }
+
+        public string PackageVersion { get; private set; }
+
+        public string PackageSourceUrl { get; private set; }
+
+        public HashSet<string> Usings { get; private set; }
+
+        private AdditionalAssembly()
+        {
+        }
+
+        protected bool Equals(AdditionalAssembly other)
+        {
+            return string.Equals(AssemblyName, other.AssemblyName, StringComparison.OrdinalIgnoreCase)
+                   && string.Equals(AssemblyPath, other.AssemblyPath, StringComparison.OrdinalIgnoreCase)
+                   && string.Equals(PackageName, other.PackageName, StringComparison.OrdinalIgnoreCase)
+                   && string.Equals(PackageVersion, other.PackageVersion, StringComparison.OrdinalIgnoreCase)
+                   && string.Equals(PackageSourceUrl, other.PackageSourceUrl, StringComparison.OrdinalIgnoreCase)
+                   && Equals(Usings, other.Usings);
```

Comment:
That also compare using just the instances, not the values in them, no?
You need `SetEquals` here, IIRC

Reply:
Fixed.